File: refine_capitalization.js
import fs from 'fs';
import path from 'path';

const rootDir = 'c:/Users/Zero/Desktop/AuraMarket/web';
const extensions = ['.js', '.jsx', '.css'];

const patterns = [
    { regex: /\buppercase tracking-widest\b/g, replacement: 'tracking-wide' },
    { regex: /\buppercase tracking-wider\b/g, replacement: 'tracking-normal' },
    { regex: /\buppercase tracking-tight\b/g, replacement: 'tracking-tight' },
    { regex: /\buppercase\b/g, replacement: '' },
    { regex: /\bBUY NOW\b/g, replacement: 'Buy Now' },
    { regex: /\bOUT OF STOCK\b/g, replacement: 'Out of Stock' },
    { regex: /\bSOLD OUT\b/g, replacement: 'Sold Out' },
    { regex: /\bLOGIN\b/g, replacement: 'Login' },
    { regex: /\bSIGN OUT\b/g, replacement: 'Sign Out' },
    { regex: /\bDISCOVERY\b/g, replacement: 'Discovery' },
];

function processFile(filePath) {
    let content = fs.readFileSync(filePath, 'utf8');
    let newContent = content;

    for (const { regex, replacement } of patterns) {
        newContent = newContent.replace(regex, replacement);
    }

    if (newContent !== content) {
        fs.writeFileSync(filePath, newContent, 'utf8');
        return true;
    }
    return false;
}

function walkDir(dir) {
    const files = fs.readdirSync(dir);
    let count = 0;

    for (const file of files) {
        const filePath = path.join(dir, file);
        const stat = fs.statSync(filePath);

        if (stat.isDirectory()) {
            if (file !== 'node_modules' && file !== '.next') {
                count += walkDir(filePath);
            }
        } else if (extensions.includes(path.extname(file))) {
            if (processFile(filePath)) {
                count++;
            }
        }
    }
    return count;
}

const total = walkDir(rootDir);
console.log(`Processed ${total} files.`);

File: refine_capitalization.py
import os
import re

root_dir = r'c:\Users\Zero\Desktop\AuraMarket\web'
extensions = ('.js', '.jsx', '.css')

patterns = [
    (r'\buppercase tracking-widest\b', 'tracking-wide'),
    (r'\buppercase tracking-wider\b', 'tracking-normal'),
    (r'\buppercase tracking-tight\b', 'tracking-tight'),
    (r'\buppercase\b', ''),
    (r'\bBUY NOW\b', 'Buy Now'),
    (r'\bOUT OF STOCK\b', 'Out of Stock'),
    (r'\bSOLD OUT\b', 'Sold Out'),
    (r'\bLOGIN\b', 'Login'),
    (r'\bSIGN OUT\b', 'Sign Out'),
    (r'\bDISCOVERY\b', 'Discovery'),
]

def process_file(filepath):
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()
    
    new_content = content
    for pattern, replacement in patterns:
        new_content = re.sub(pattern, replacement, new_content)
    
    if new_content != content:
        with open(filepath, 'w', encoding='utf-8') as f:
            f.write(new_content)
        return True
    return False

count = 0
for subdir, dirs, files in os.walk(root_dir):
    if 'node_modules' in subdir or '.next' in subdir:
        continue
    for file in files:
        if file.endswith(extensions):
            if process_file(os.path.join(subdir, file)):
                count += 1

print(f"Processed {count} files.")
